server/conn.go

```diff
@@ -954,19 +954,24 @@ func (cc *clientConn) handleLoadData(ctx context.Context, loadDataInfo *executor
 	}
 	loadDataInfo.SetMessage()
 
-	if err = loadDataInfo.Ctx.StmtCommit(); err != nil {
-		return errors.Trace(err)
-	}
-	txn, err := loadDataInfo.Ctx.Txn(true)
 	if err != nil {
+		loadDataInfo.Ctx.StmtRollback()
+	} else {
+		err = loadDataInfo.Ctx.StmtCommit()
+	}
+
+	if txn, err1 := loadDataInfo.Ctx.Txn(true); err1 == nil {
 		if txn != nil && txn.Valid() {
-			if err1 := txn.Rollback(); err1 != nil {
-				log.Errorf("load data rollback failed: %v", err1)
+			if err != nil {
+				if err1 := txn.Rollback(); err1 != nil {
+					log.Errorf("load data rollback failed: %v", err1)
+				}
+				return errors.Trace(err)
 			}
 		}
-		return errors.Trace(err)
+	} else {
+		log.Error(err1)
```

Comment:
Please add some information like line967.
And If `err1 != nil` is true, I think we can't commit this transaction.

Reply:
Maybe panic here will be better.

Reply:
done.